explorer/src/context/gateway.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { ApiState, GatewayReportResponse, UptimeStoryResponse } from '../typeDefs/explorer-api';
+import { Api } from '../api';
+import { useApiState } from './hooks';
+
+/**
+ * This context provides the state for a single gateway by identity key.
+ */
+
+interface GatewayState {
+  uptimeReport?: ApiState<GatewayReportResponse>;
+  uptimeStory?: ApiState<UptimeStoryResponse>;
+}
+
+export const GatewayContext = React.createContext<GatewayState>({});
+
+export const useGatewayContext = (): React.ContextType<typeof GatewayContext> =>
+  React.useContext<GatewayState>(GatewayContext);
+
+interface GatewayContextProviderProps {
+  gatewayIdentityKey: string;
+}
+
+/**
+ * Provides a state context for a gateway by identity
+ * @param gatewayIdentityKey   The identity key of the gateway
+ */
+export const GatewayContextProvider: React.FC<GatewayContextProviderProps> = ({ gatewayIdentityKey, children }) => {
```

Comment:
Please, in general, avoid using  `React.FC` -> https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components